sparse/mlir_backend/_constructors.py

```diff
@@ -216,7 +215,7 @@ def to_sps(self) -> sps.csr_array:
                 pos = self.pos.to_numpy()[: self.pos_len]
                 crd = self.crd.to_numpy()[: self.crd_len]
                 data = self.data.to_numpy()[: self.data_len]
-                return sps.csr_array((data, crd, pos), shape=(self.shape_x, self.shape_y))
+                return sps.csr_array((data.copy(), crd.copy(), pos.copy()), shape=(self.shape_x, self.shape_y))
```

Comment:
And we do a double copy here... Once in `to_numpy` as well.

Reply:
Closing as we discussed it in a call.